[wendyng0107] iP

wendy0107 opened the pull request:
# Duke
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))
- text-based
-  beginner-friendly
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is:

1.  Download it from [here](https://github.com/wendy0107/ip/releases/tag/A-Jar).
2.  Double-click it.
3.  Add your tasks.
4.  Let Duke manage your tasks for you :stuck_out_tongue_winking_eye:

Plus it's **FREE**!

### Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you're a Java programmer, you can also use it to practice Java too! Here's the `main` method:
```ruby
public static void main(String[] args) {
    new Duke(PATH).run();
}
```

Commit:
Add Gradle support

wendy0107 committed:
Implemented Level 1: Greet, Echo, Exit

wendy0107 committed:
Implemented Add and List

wendy0107 committed:
Implemented Level 3: Mark as Done

wendy0107 committed:
Implemented Level 4: Todo, Event, Deadline

wendy0107 committed:
Added A-TextUiTesting

wendy0107 committed:
Created some exceptions

wendy0107 committed:
Fixed typos

wendy0107 committed:
Modified classes to handle exceptions and fixed minor typos

wendy0107 committed:
Implemented Delete

wendy0107 committed:
Updated to handle delete exceptions

wendy0107 committed:
Specify import statements

wendy0107 committed:
Create Storage class to support saving data

wendy0107 committed:
Add methods to Task, Deadline, Event and Todo to support reading files

wendy0107 committed:
Add methods to TaskList to support reading files and storing data in files

wendy0107 committed:
Implement Level 7 - saving and reading data from files

wendy0107 committed:
Modify DukeException to handle exceptions related to files

wendy0107 committed:
Deleted repeat and add from Duke

wendy0107 committed:
Implemented Level 8 - taught Duke to understand dates

wendy0107 committed:
Modify DukeException to handle errors with date formats

wendy0107 committed:
Merge branch 'branch-Level-8' - resolved conflict in DukeException

wendy0107 committed:
Modify toFile method to support LocalDate format

wendy0107 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

wendy0107 committed:
Extract Ui and Command classes

wendy0107 committed:
Extract Parser classes

wendy0107 committed:
Implement Ui, Parser and Command class methods

wendy0107 committed:
Organise classes into packages duke, duke.command and duke.task

wendy0107 committed:
Fix technical problem from merging gradle

wendy0107 committed:
Add test methods for Todo, Deadline and Event class

wendy0107 committed:
Add test methods in DeadlineTest

wendy0107 committed:
Update Storage to handle missing directory in file path

wendy0107 committed:
Added Java documentation.

wendy0107 committed:
Add Find command and modify Parser, TaskList and DukeException to handle find command.

wendy0107 committed:
Edit code to follow coding standard

wendy0107 committed:
Merge branch 'branch-Level-9'

wendy0107 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/TaskList.java

wendy0107 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/DukeException.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/Todo.java

wendy0107 committed:
Add documentation to TaskList for level 9

wendy0107 committed:
Add checkstyle

wendy0107 committed:
Merge branch 'branch-A-CheckStyle'

NBQian commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,64 @@
+     * @param command type of exception thrown.
+     */
+    public DukeException(String command) {
+        if (commandError.contains(command)) {
```
I like how there is no deep nesting here which makes the code very readable. However, maybe you can consider using switch statements which makes it even more readable.

wendy0107 replied:
Yeah I agree, thank you, I'll make the necessary changes!

NBQian commented on src/main/java/duke/command/ToDoCommand.java:
```diff
@@ -0,0 +1,22 @@
+import duke.Ui;
+import duke.task.Todo;
+
+public class ToDoCommand extends Command {
```
not compulsory, but maybe it's good to add javadoc for this and other public classes

wendy0107 replied:
Good point! I'll add them, thank you!

NBQian commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,64 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/** Class contains all exceptions thrown by Duke. */
```
I like that you are wrapping your lines so that each line is not too long.

wendy0107 replied:
Thank you!

NBQian commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,187 @@
+        LocalDate endDate;
+        Command command = null;
+
+        switch (commandType) {
```
I like the use of switch statements here, very easy to read your code

wendy0107 replied:
Thank you for the feedback!

NBQian commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,163 @@
+/**
+ * Class contains variables and methods related to modifying a list of Tasks.
+ */
+public class TaskList {
```
Maybe you can name lst as something more meaningful, say, "taskList".

wendy0107 replied:
Yeah I agree! I'll change it, thank you!

Bransonlj commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,83 @@
+     * @param lst TaskList file data is to be laoded into.
+     * @throws DukeException If file does not exist.
+     */
+    public void loadFileInto(TaskList lst) throws DukeException {
```
Perhaps you can name lst something more meaningful like "taskList".

wendy0107 replied:
Yeah I agree! I'll make the necessary changes thank you!

Bransonlj commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,155 @@
+ * Class contains variables and methods related to interactions with the user.
+ */
+public class Ui {
+    private static final String line = "____________________________________________________________";
```
Maybe can name it in all caps "LINE" since its a constant!

wendy0107 replied:
I'll do that! Thank you!

Bransonlj commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,22 @@
+import duke.Ui;
+import duke.task.Deadline;
+
+public class DeadlineCommand extends Command {
```
maybe can add javadoc to say what each command subclass is for for better clarity!

wendy0107 replied:
Good point! I'll add it in, thank you!

Bransonlj commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,76 @@
+ * Abstract class contains variables and methods related to Tasks.
+ */
+public abstract class Task {
+    protected boolean completed;
```
could name it isCompleted, so it sounds more like a boolean!

wendy0107 replied:
I'll change that, thank you!

Bransonlj reviewed (commented):
Looks very neat and nice! only managed to find a few stuff

wendy0107 committed:
Improve code quality after Peer Review

wendy0107 committed:
Added GUI functinality to Duke

wendy0107 committed:
Merge branch 'branch-Level-10'

wendy0107 committed:
Update Javadocs and removed text based Ui

wendy0107 committed:
Add assert statements and handle out of bounds error in delete

wendy0107 committed:
Improve code quality

Several methods in the Parser class exceed 30 LoC and there is common
behavior between the cases which causes code duplication.
Remaining unused methods from the text-based Ui implementation may
trip up readers.

Extracting this common behaviour allows us to improve readability
of the code by reducing code duplication.

Checking the validity and sufficiency of arguments is extracted into
new methods. Some methods are also reorganised to make the code
flow more logically. Unused methods from the Task and Ui class have also
been removed.

wendy0107 committed:
Merge pull request #2 from wendy0107/branch-A-Assertions

Add assert statements

wendy0107 committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/duke/TaskList.java

wendy0107 committed:
Implemented Extension B: ViewSchedule

Users are able to add deadlines and events with specified dates

However, to view a list, users can only choose between 'list' to view
every single task or 'find' which only shows tasks with matching names.
Hence, adding a view schedule functionality will give the users more
flexibility.

Add the supporting methods to TaskList, Task and Parser and Ui.
DukeException has additional cases added to handle errors by 'view'.

This gives the users more flexibility in using their list by supporting
the need to view their schedules for a specific date.

wendy0107 committed:
Merge pull request #4 from wendy0107/branch-BCD-Extension

Implemented Extension B: ViewSchedule

wendy0107 committed:
Refactor Duke to Luffy and updated Java Docs

wendy0107 committed:
Add Ui.png and update View command

wendy0107 committed:
Updated Ui.png and fix text overrun in DialogBox

wendy0107 committed:
Update User Guide

wendy0107 committed:
Fix Main class path in build gradle for Jar release

wendy0107 committed:
Update README.md
